tests/linearizability/failpoints.go

```diff
@@ -75,6 +77,58 @@ type Failpoint interface {
 	Name() string
 }
 
+type sleepFailpoint struct {
+	failpoint string
+	payload   string
+}
+
+func (sf sleepFailpoint) Name() string {
+	return sf.failpoint
+}
+
+func (sf sleepFailpoint) Trigger(ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	address := fmt.Sprintf("127.0.0.1:%d", member.Config().GoFailPort)
+	err := setupSleepFailpoint(address, sf.failpoint, sf.payload)
+	if err != nil {
+		return fmt.Errorf("sleep gofailpoint setup failed: %w", err)
+	}
+	return nil
+}
+
+func setupSleepFailpoint(host, failpoint, payload string) error {
+	//payload should be in the form of: 'sleep(600)'
+	err := setupGoFailpoint(host, failpoint, payload)
+	if err != nil {
+		return err
+	}
+	failpointsUrl := url.URL{
+		Scheme: "http",
+		Host:   host,
+		Path:   failpoint,
+	}
+	//check whether sleep was enabled
+	r, err := http.NewRequest("GET", failpointsUrl.String(), nil)
```

Comment:
Don't think this is needed. If `setupGoFailpoint` succeeded then we can assume that sleep was enabled.